Explain experiments and use case recipe in docs (#143)
phenology/springtime: docs/experiments.md

<!--
SPDX-FileCopyrightText: 2023 Springtime authors

SPDX-License-Identifier: Apache-2.0
-->

We provide several "recipes" for downloading [data from various
sources](https://springtime.readthedocs.io/en/latest/datasets/) and executing a
range of [machine learning
methods](https://springtime.readthedocs.io/en/latest/models/) with it.

> Note that a "recipe" is a file with `yaml` extension.

Reviewer: Extension does not matter for springtime cli, the format does.
```suggestion
Note that a "recipe" is a file in [yaml format](https://yaml.org/).
```
